addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js

```diff
@@ -221,22 +232,61 @@ export const editorCommands = {
         // element if it's a block then we insert the content in the right places.
         let currentNode = startNode;
         let lastChildNode = false;
+        const currentList = currentNode && closestElement(currentNode, 'UL, OL');
+        const mode = currentList && getListMode(currentList);
+        const convertList = (node) => {
+            const currNode = node.firstChild;
+            const startNode = firstLeaf(node);
+            const endNode = lastLeaf(node);
+            editor.historyPauseSteps();
+            setSelection(startNode, 0, endNode, nodeSize(endNode));
+            const restoreCursor = preserveCursor(editor.document);
+            const toConvert = closestElement(startNode, 'ul, ol');
+            if (mode !== getListMode(toConvert)) {
+                editor._applyCommand('toggleList', mode);
+            } else if (node.firstChild && (mode === getListMode(node) || node.classList.contains('oe-nested'))) {
+                const childNodes = node.childNodes;
+                [...childNodes].forEach(n => n.classList.contains('oe-nested') && convertList(n));
+            }
+            restoreCursor();
+            document.getSelection().collapseToEnd();
+            editor.historyUnpauseSteps();
+            return currNode.parentNode;
+        }
         const _insertAt = (reference, nodes, insertBefore) => {
             for (const child of (insertBefore ? nodes.reverse() : nodes)) {
                 reference[insertBefore ? 'before' : 'after'](child);
                 reference = child;
             }
         }
+        let convertedList;
         if (containerLastChild.hasChildNodes()) {
             const toInsert = [...containerLastChild.childNodes]; // Prevent mutation
             _insertAt(currentNode, [...toInsert], insertBefore);
-            currentNode = insertBefore ? toInsert[0] : currentNode;
-            lastChildNode = toInsert[toInsert.length - 1];
+            if (tagULOL.includes(toInsert[0].nodeName)) {
+                convertedList = convertList(toInsert[0]);
+                // Pasting a list between a li, merge the right split into the
+                // pasted list by deleting forward from the end of the pasted list.
+                if (convertedList.nextSibling) {
+                    editor.historyPauseSteps();
+                    editor._applyCommand('oDeleteForward');
+                    editor.historyUnpauseSteps();
+                }
+            }
+            currentNode = insertBefore ?
+                (editor.editable.contains(toInsert[0]) && toInsert[0]) || convertedList :
+                currentNode;
+            lastChildNode = editor.editable.contains(toInsert[toInsert.length - 1]) ? toInsert[toInsert.length - 1] : convertedList;
```

Comment:
Seeing this again I don't think I had to check if `editor.editable` contains the node as `convertList` returns the converted node then it would be simply `convertedList || toInsert[0]`.